lib/actions/parse.js

```diff
@@ -0,0 +1,261 @@
+"use strict";
+
+var util = require("util");
+var _ = require("lodash/fp");
+var objMap = require("lodash/map"); // TODO GLOBAL: Unwind this with global config.
+var parse = require("babylon").parse;
+var traverse = require("babel-traverse").default;
+var t = require("babel-types");
+
+var Base = require("./base");
+var metaSum = require("../utils/data").metaSum;
+var toParseable = require("../utils/code").toParseable;
+
+
+function isExports (node) {
+  return t.isIdentifier(node) && t.name === "exports";
+}
+
+function isModuleExports(node) {
+  return t.isMemberExpression(node) &&
+    t.isIdentifier(node.object) &&
+    t.isIdentifier(node.property) &&
+    node.object.name === "module" &&
+    node.property.name === "exports";
+}
+
+function isExportsSubproperty (node) {
+  if (!node.object) { return false; }
+
+  return isExports(node.object) ||
+    isModuleExports(node.object) ||
+    // module.exports.foo.something
+    isExportsSubproperty(node.object);
+}
+
+function isWebpackRequire (node) {
+  return t.isCallExpression(node) &&
+    t.isIdentifier(node.callee) &&
+    node.callee.name === "__webpack_require__";
+}
+
+function getReExportedModules (moduleAst) {
+  var otherModuleReferences = Object.create(null);
+
+  traverse(moduleAst, {
+    noScope: true,
+    AssignmentExpression: function (path) {
+      var node = path.node;
+
+      if (
+      // module.exports.foo.something = __webpack_require__(4);
+      // module.exports.foo = __webpack_require__(4);
+        isExportsSubproperty(node.left) &&
+        isWebpackRequire(node.right)
+      ) {
+        otherModuleReferences[node.right.arguments[0].value] = true;
+      } else if (
+      // module.exports = { foo: __webpack_require__(4) };          
+        t.isObjectExpression(node.right) &&
+        isModuleExports(node.left)
+      ) {
+        _.each(function (objectProperty) {
+          if (isWebpackRequire(objectProperty.value)) {
+            otherModuleReferences[objectProperty.value.arguments[0].value] = true;
+          }
+        })(node.right.properties);
+      }
+    }
+
+  });
+
+  // Don't report two re-exported values if they both come from the same
+  // imported module.
+  return _.keys(otherModuleReferences).length;
+}
+
+/**
+ * Suspect parse functions.
+ *
+ * For `--suspect-parses`.
+ */
+var SUSPECT_PARSES = {
+  // Multiple Exports.
+  //
+  // ```js
+  // // Single declaration
+  // module.exports = {
+  //   foo: __webpack_require__(1),
+  //   bar: __webpack_require__(2)
+  // }
+  //
+  // // Multiple declaration
+  // module.exports.foo = __webpack_require__(1);
+  // module.exports.bar = __webpack_require__(2);
+  // ```
+  MULTIPLE_EXPORT: function (src) {
+    var ast = parse(src, { sourceType: "module" });
+    return getReExportedModules(ast) > 1;
```

Comment:
@ryan-roemer, it appears as if the template below expects the return value here to be a string if truthy.  Since exports could be defined in different places, I wasn't clear on how to proceed.  Will leave that up to you to resolve.